sdk/program/src/lib.rs

```diff
@@ -636,6 +636,35 @@ pub mod vote {
     }
 }
 
+/// A vector of all Solana SDK Program IDs
```

Comment:
nit: Even if we separate out `sysvar::ALL_IDS`, these aren't exactly all programs. Maybe drop `program` and just have the mod be `ids` and function be `sdk_ids` ?